pkg/datarepair/checker/checker.go

```diff
@@ -120,12 +122,39 @@ func lookupResponsesToNodes(responses *pb.LookupResponses) []*pb.Node {
 	return nodes
 }
 
-// Run
-func (c *checker) Run() error {
-	return nil
+// Run the checker loop
+func (c *checker) Run(ctx context.Context, interval time.Duration) error {
+	zap.S().Info("Checker is starting up")
+	ticker := time.NewTicker(interval)
+	defer ticker.Stop()
+	ctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	var errs []error
+
+	go func() {
+		for {
+			select {
+			case <-ticker.C:
+				zap.S().Info("Starting segment checker service")
+				go func() {
+					err := c.IdentifyInjuredSegments(ctx)
+					if err != nil {
+						errs = append(errs, err)
+					}
+				}()
+				return
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+	//TODO collectErrors and wait?
+	return utils.CombineErrors(errs...)
 }
 
-// Stop
-func (c *checker) Stop() error {
+// Stop the checker loop
+func (c *checker) Stop(ctx context.Context) error {
+	_, cancel := context.WithCancel(ctx)
+	cancel()
```

Comment:
this won't do anything. this creates a child context. The cancel only cancels the child not the parent. 

You can add a cancel func to the checker struct and then call `c.cancel()`

Reply:
gotcha I was concerned that we didn't want context on the checker struct